app/src/main/java/com/hedvig/app/feature/keygear/ui/itemdetail/KeyGearItemDetailViewModel.kt

```diff
@@ -4,24 +4,58 @@ import android.net.Uri
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.hedvig.android.owldroid.graphql.KeyGearItemQuery
+import com.hedvig.android.owldroid.type.MonetaryAmountV2Input
+import com.hedvig.app.feature.keygear.data.KeyGearItemsRepository
 import com.hedvig.app.util.LiveEvent
+import kotlinx.coroutines.launch
+import org.threeten.bp.YearMonth
 
 abstract class KeyGearItemDetailViewModel : ViewModel() {
     abstract val data: LiveData<KeyGearItemQuery.KeyGearItem>
 
     abstract val isUploading: LiveEvent<Boolean>
 
+    val purchaseDate = MutableLiveData<YearMonth>()
+
+    abstract fun choosePurchaseDate(yearMonth: YearMonth)
     abstract fun loadItem(id: String)
     abstract fun uploadReceipt(uri: Uri)
+    abstract fun updatePurchaseDateAndPrice(
+        id: String,
+        yearMonth: YearMonth,
+        price: MonetaryAmountV2Input
+    )
 }
 
-class KeyGearItemDetailViewModelImpl : KeyGearItemDetailViewModel() {
+class KeyGearItemDetailViewModelImpl(private val repository: KeyGearItemsRepository) :
+    KeyGearItemDetailViewModel() {
     override val data = MutableLiveData<KeyGearItemQuery.KeyGearItem>()
+
     override val isUploading = LiveEvent<Boolean>()
 
+    override fun choosePurchaseDate(yearMonth: YearMonth) {
```

Comment:
I think this function could be a part of the base abstract class, as it will always be the same whether or not you are mocking network calls 🙂

Reply:
Come to think of it, I think this code should actually be in its own ViewModel - ViewModels are not shared between activities.